packages/commons/src/auth/token/interopTokenGenerator.ts

```diff
@@ -1,47 +1,52 @@
 /* eslint-disable max-params */
 import { Algorithm, JwtHeader, JwtPayload } from "jsonwebtoken";
+import {
+  tokenGenerationError,
+  missingRSAKeyError,
+} from "pagopa-interop-models";
 import { v4 as uuidv4 } from "uuid";
+import { logger, signerConfig } from "../../index.js";
 import { userRoles } from "../authData.js";
-import { buildPrivateKeysKidHolder } from "../keys/keyHolder.js";
 import { buildSignerService } from "../signerService.js";
-import { logger } from "../../index.js";
-import { jwtSeedConfig } from "../../config/jwtConfig.js";
-import { InternalToken, TokenSeed } from "./token.js";
+import { InternalToken, TokenSeed, TokenPayloadSeed } from "./token.js";
 
 export type InteropTokenGenerator = {
-  generateInternalToken: () => Promise<InternalToken>;
+  generateInternalToken: (seed: TokenPayloadSeed) => Promise<InternalToken>;
 };
 
-const createInternalTokenWithKid = (
+const createInternalToken = (
   algorithm: Algorithm,
   kid: string,
   subject: string,
   audience: string[],
   tokenIssuer: string,
   validityDurationSeconds: number
 ): TokenSeed => {
-  const issuedAt = new Date();
+  const issuedAt = new Date().getTime() / 1000;
+  const expireAt = validityDurationSeconds * 1000 + issuedAt;
+
   return {
     id: uuidv4(),
     algorithm,
     kid,
     subject,
     issuer: tokenIssuer,
-    issuedAt: issuedAt.getTime() / 1000,
-    nbf: issuedAt.getTime() / 1000,
-    expireAt: issuedAt.setSeconds(
-      issuedAt.getSeconds() + validityDurationSeconds
-    ),
+    issuedAt,
+    nbf: issuedAt,
+    expireAt,
     audience,
     customClaims: new Map([["role", userRoles.INTERNAL_ROLE]]),
   };
 };
 
 export const buildInteropTokenGenerator = (): InteropTokenGenerator => {
   // Hosting all the dependencies to collect all process env reading at one time
-  const kidHolder = buildPrivateKeysKidHolder();
   const signerService = buildSignerService();
-  const { subject, audience, tokenIssuer, secondsToExpire } = jwtSeedConfig();
+  const { rsaKeysIdentifiers } = signerConfig();
+
+  if (!rsaKeysIdentifiers.length) {
+    throw missingRSAKeyError();
+  }
```

Comment:
is it possible to move this validation in [zod definition](https://github.com/pagopa/interop-be-monorepo/pull/225/files/82fd4ea61ee68a880b04029c5bb3e03a3a1d4993..b11b0060ffebbc55c37309796994dfb155d57206#diff-aba56c01a025ea369b69799c4b225dbb74d3270a4e9595809ca37e160374d026R7)?
So this is clear from the data model and does not need to be replicated in future logics

Reply:
A few consideration
I'm agree to using Zod [nonempty() validation](https://zod.dev/?id=nonempty) when keys are parsed from env variables, in this case I proceed to remove `missingRSAKeyError` error type.